cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/ComputerVision.scala

```diff
@@ -261,22 +266,36 @@ trait BasicAsyncReply extends HasAsyncReply {
       blocking {
         Thread.sleep(getInitialPollingDelay.toLong)
       }
-      val it = (0 to maxTries).toIterator.flatMap { i =>
-        queryForResult(key, client, location).orElse({
-          blocking {
-            Thread.sleep(getPollingDelay.toLong)
+      val it = (0 to maxTries).toIterator.foldLeft[Either[String, HTTPResponseData]](Left(""))( (prevResponse, i) => {
+        prevResponse match {
+          case Right(value) => Right(value) // have value don't want to requery
+          case Left(value) => {
+            if (i > 0){
+              blocking {
+                Thread.sleep(getPollingDelay.toLong)
+              }
+            }
+            queryForResult(key, client, location)
+          }
+        }
+      })
+      it match {
+        case Right(value) =>{
+          val asyncCompletionLogMessagePrefixValue = getAsyncCompletionLogMessagePrefix
+          if (asyncCompletionLogMessagePrefixValue != "") {
```

Comment:
Dont need to use "" to connote param not being set. get(paramName) will return none if param isn't set and can be used to turn off setting. However, it might be good to ask the broader existential question of why a user would need to set the log prefix manually. Are you trying to parse logs in a unified way? If so perhaps we can take a page from the messages output in logClass and logTransform and output a nice json payload that log parsers can inspect and display reasonably

Reply:
Simplified as we discussed